doc/adr/0010-reducing-snapshot-size.md

```diff
@@ -0,0 +1,409 @@
+        "A": 20,
+        "B": 10,
+        "C": 10
+      }
+    }
+  }
+}
+```
+
+#### Snapshot 2
+
+```json
+{
+  "ordinal": 2,
+  "blocks": [
+    {
+      "transactions": [
+        {
+          "source": "C",
+          "destination": "D",
+          "amount": 10
+        }
+      ]
+    }
+  ],
+  "infos": {
+    "snapshot": {
+      "balances": {
+        "C": 0,
+        "D": 10
+      }
+    },
+    "aggregated": {
+      "since": {
+        "ordinal": 1,
+        "hash": "3c12..."
+      },
+      "balances": {
+        "B": 10,
+        "C": 0,
+        "D": 10
+      }
+    }
+  }
+}
+```
+
+#### Snapshot 1000
+
+```json
+{
+  "ordinal": 1000,
+  "blocks": [
+    {
+      "transactions": [
+        {
+          "source": "D",
+          "destination": "E",
+          "amount": 10
+        }
+      ]
+    }
+  ],
+  "infos": {
+    "snapshot": {
+      "balances": {
+        "D": 0,
+        "E": 10
+      }
+    },
+    "aggregated": {
+      "since": {
+        "ordinal": 1,
+        "hash": "3c12..."
+      },
+      "balances": {
+        "B": 10,
+        "C": 0,
+        "D": 0,
+        "E": 10
+      }
+    }
+  }
+}
+```
+
+#### Snapshot 1001
+
+```json
+{
+  "ordinal": 1001,
+  "blocks": [
+    {
+      "transactions": [
+        {
+          "source": "E",
+          "destination": "F",
+          "amount": 10
+        }
+      ]
+    }
+  ],
+  "infos": {
+    "snapshot": {
+      "balances": {
+        "E": 0,
+        "F": 10
+      }
+    },
+    "aggregated": {
+      "since": {
+        "ordinal": 0,
+        "hash": "8f17..."
+      },
+      "balances": {
+        "A": 20,
+        "B": 10,
+        "C": 0,
+        "D": 0,
+        "E": 0,
+        "F": 10
+      }
+    }
+  }
+}
+```
+
+#### Snapshot 1002
+
+```json
+{
+  "ordinal": 1002,
+  "blocks": [
+    {
+      "transactions": [
+        {
+          "source": "F",
+          "destination": "G",
+          "amount": 10
+        }
+      ]
+    }
+  ],
+  "infos": {
+    "snapshot": {
+      "balances": {
+        "F": 0,
+        "G": 10
+      }
+    },
+    "aggregated": {
+      "since": {
+        "ordinal": 1001,
+        "hash": "93ac..."
+      },
+      "balances": {
+        "E": 0,
+        "F": 0,
+        "G": 10
+      }
+    }
+  }
+}
+```
+
+#### Snapshot 2000
+
+```json
+{
+  "ordinal": 2000,
+  "blocks": [],
+  "infos": {
+    "snapshot": {},
+    "aggregated": {
+      "since": {
+        "ordinal": 1001,
+        "hash": "93ac..."
+      },
+      "balances": {
+        "E": 0,
+        "F": 0,
+        "G": 10
+      }
+    }
+  }
+}
+```
+
+#### Snapshot 2001
+
+```json
+{
+  "ordinal": 2001,
+  "blocks": [],
+  "infos": {
+    "snapshot": {},
+    "aggregated": {
+      "since": {
+        "ordinal": 0,
+        "hash": "8f17..."
+      },
+      "balances": {
+        "A": 10,
+        "B": 10,
+        "C": 0,
+        "D": 0,
+        "E": 0,
+        "F": 0,
+        "G": 10
+      }
+    }
+  }
+}
+```
+
+
+## Consequences
```

Comment:
there can be also positive consequences: I'd also say that a consequence of that decision is ability to scan (by a block explorer, or any other network scanner) partial data and update the state partially.